### PR TITLE
Logging infrastructure update

### PR DESCRIPTION
**Pull request by sean-gilliam**
This PR replaces all references to the old `RS.Bug`/`RS.Log` to the new `CLogger` methods.

Things to note:
- Removed the `fmt` library since `spdlog` already bundled it. All references switched to the one under `spdlog`.

- Tied `errno` messages previously displayed by `perror` to `std::strerror(errno)` so those messages flow through our logging infra.

- `bugout`/`bugexit` still exists for the time being so we can come up with a logging plan for them.

Sorry for this extremely hefty PR. I broke it out over several commits to help digest it more easily. Let me know if you have questions/concerns.

### CODE REVIEW

**Comment by sean-gilliam** [comment]
Here's what the console looks like on error.

rs-console-error

Here's what the console looks like with a clean run.
rs-console good

**Comment by sean-gilliam** [comment]
Here's what the log file shows on a clean run

```
[2024-03-09 00:05:23.427] [rs] [info] *** Beginning RIFTSHADOW MUD server ***
[2024-03-09 00:05:23.427] [rs] [info] Creating persistent SQL connection...
[2024-03-09 00:05:23.428] [rs] [info] Loading options and greeting screen...
[2024-03-09 00:05:23.428] [rs] [info] Loading class and race data...
[2024-03-09 00:05:23.428] [rs] [info] Initialize random number generator...
[2024-03-09 00:05:23.428] [rs] [info] Setting time, weather, and berus/calabren pos...
[2024-03-09 00:05:23.428] [rs] [info] Assigning gsns and psns to skills and spells...
[2024-03-09 00:05:23.428] [rs] [info] Reading the race info...
[2024-03-09 00:05:23.428] [rs] [info] Sorting area list..
[2024-03-09 00:05:23.428] [rs] [info] Loading area files..
[2024-03-09 00:05:23.429] [rs] [info] Loading area files now...
[2024-03-09 00:05:23.533] [rs] [info] Starting player file parsing..
[2024-03-09 00:05:23.533] [rs] [info] Loading object counts off players now...
[2024-03-09 00:05:23.534] [rs] [info] Object Limits loaded
[2024-03-09 00:05:23.534] [rs] [info] Cabal Limits loaded
[2024-03-09 00:05:23.538] [rs] [info] Exits fixed
[2024-03-09 00:05:23.558] [rs] [info] Adjacent areas found
[2024-03-09 00:05:23.558] [rs] [info] Notes cleaned
[2024-03-09 00:05:23.580] [rs] [info] Area update
[2024-03-09 00:05:23.581] [rs] [info] Allocating gold
[2024-03-09 00:05:23.581] [rs] [info] Cabal Items loaded
[2024-03-09 00:05:23.581] [rs] [info] Priming weather
[2024-03-09 00:05:23.581] [rs] [info] Riftshadow booted, binding on port 9999.
```

**Comment by sean-gilliam** [comment]
We can update the logging pattern later. This was just the bare-bones first attempt to get something together.

File: code/devextra.c [hunk 1]
```diff
@@ -170,7 +170,7 @@ void clean_mud()
 	if (!nSQL.StartSQLServer(riftCore.Host.c_str(),
 		riftCore.Db.c_str(), riftCore.User.c_str(), riftCore.Pwd.c_str()))
 	{
-		RS.Bug("clean_mud: failed to establish a database connection.");
+		RS.Logger.Warn("clean_mud: failed to establish a database connection.");
```
Comment (Psypher9) [inline code/devextra.c:173]: Do you feel like this might be better as an error?

File: code/sql.c [hunk 2]
```diff
@@ -30,7 +30,7 @@ bool CSQLInterface::StartSQLServer(const char* host, const char* db, const char*
 
 	if(!mysql_real_connect(connection, host, user, pass, db, 0, nullptr, 0))
 	{
-		RS.Bug("Unable to connect to mysql database: %s\r\n", mysql_error(connection));
+		RS.Logger.Warn("Unable to connect to mysql database: {}\r\n", mysql_error(connection));
```
Comment (Psypher9) [inline code/sql.c:33]: Would this be better as an error as well?

File: code/sql.c [hunk 3]
```diff
@@ -54,7 +54,7 @@ int CSQLInterface::IQuery(const char *query)
 	
 	if(mysql_query(connection, query))
 	{
-		RS.Bug("MYSQL IQuery Error: %s on query %s", mysql_error(connection), query);
+		RS.Logger.Warn("MYSQL IQuery Error: {} on query {}", mysql_error(connection), query);
```
Comment (Psypher9) [inline code/sql.c:57]: This maybe could be an error too? I'm not 100% on this so whatever you think

File: code/sql.c [hunk 4]
```diff
@@ -67,7 +67,7 @@ int CSQLInterface::IQuery(const char *query)
 	else if(mysql_field_count(connection) == 0)
 		return mysql_affected_rows(connection);
 	else
-		RS.Bug("MYSQL IQuery Store Error: %s on query %s", mysql_error(connection), query);
+		RS.Logger.Warn("MYSQL IQuery Store Error: {} on query {}", mysql_error(connection), query);
```
Comment (Psypher9) [inline code/sql.c:70]: Possibly an error log here too?

File: code/update.c [hunk 5]
```diff
@@ -2297,8 +2298,8 @@ void affect_update(void)
 
 			wiznet("ERROR: $N had corrupt name!  Name has been properly repaired.  Check for other errors.", ch, nullptr, 0, 0, 0);
 			sprintf(buf, "Error: %s has corrupt name.  Repaired.\n\r", ch->true_name);
-			RS.Bug(buf);
-			RS.Log(buf);
+			RS.Logger.Debug(buf);
```
Comment (Psypher9) [inline code/update.c:2301]: Do we need both a debug and Info here?

**Review by Psypher9 — CHANGES_REQUESTED** [review]

**Comment by Psypher9** [comment]
These log screen caps look beautiful

File: code/devextra.c (hunk 1 above)
Comment (sean-gilliam) [reply code/devextra.c]: Fixed in latest commit.

File: code/sql.c (hunk 2 above)
Comment (sean-gilliam) [reply code/sql.c]: Fixed in latest commit.

File: code/sql.c (hunk 3 above)
Comment (sean-gilliam) [reply code/sql.c]: Fixed in latest commit.

File: code/sql.c (hunk 4 above)
Comment (sean-gilliam) [reply code/sql.c]: Fixed in latest commit.

File: code/update.c (hunk 5 above)
Comment (sean-gilliam) [reply code/update.c]: Fixed in latest commit.

**Comment by sean-gilliam** [comment]
> These log screen caps look beautiful

Gruvbox Material Dark theme

**Comment by rezalas** [comment]
This is awesome progress man!!! I'm looking forward to this so I can rebase and continue work on the stored procedures.

**Review by Psypher9 — APPROVED** [review]

**Pull request merged**